### DISCUSSION TITLE
Add validation mode for each validation

### DISCUSSION THREAD

**Discussion by TzGreen** [react-hook-form/react-hook-form#8331] [Q&A] [answered]
Hi,
I'm trying to set a complicated validation on fields, where each validation has its own triggering mode. 
I think it'll be easer with a use-case scenario: 

I have an input of a user name.
The validations I have for this field are: 
- required
- pattern of letters only
- validation that this username doesn't exist in my users list
- min length of 2 chars
- max length of 20 chars

Now, on the **onChange** event / mode, I'd like to check **only** if the username matches the pattern and that it isn't longer than the max length. If the pattern or max length validations don't pass I will display errors.

on the **onBlur** event / mode, I'd like to verify **only** if the username doesn't exist and the its length accedes the min length, and if not I will display errors.

Same thing with **onSubmit** event / mode, were I'll validate all of the rest.

I implemented this myself, but I'm still looking to see if there is a better way doing this.
I'll be happy for any idea.
Thanks!

**Comment by bluebill1049** [comment, marked as answer]
use trigger for custom validation behaviour.

**Reply by TzGreen** [reply]
Thank you
